source/shared/core_stmt.cpp

```diff
@@ -2218,145 +2223,195 @@ void get_field_as_string( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT field_ind
                 if( is_truncated_warning( state ) ) {
 #endif // !_WIN32  
 
-					SQLLEN dummy_field_len = 0;
-
-					// for XML (and possibly other conditions) the field length returned is not the real field length, so
-					// in every pass, we double the allocation size to retrieve all the contents.
-					if( field_len_temp == SQL_NO_TOTAL ) {
-
-						// reset the field_len_temp
-						field_len_temp = INITIAL_FIELD_STRING_LEN;
-
-						do {
-							SQLLEN initial_field_len = field_len_temp;
-							// Double the size. 
-							field_len_temp *= 2;
-
-							field_value_temp = static_cast<char*>( sqlsrv_realloc( field_value_temp, field_len_temp + extra + 1 ));
-
-							field_len_temp -= initial_field_len;
-
-							// Get the rest of the data.
-							r = stmt->current_results->get_data( field_index + 1, c_type, field_value_temp + initial_field_len,
-								field_len_temp + extra, &dummy_field_len,
-								false /*handle_warning*/ TSRMLS_CC );
-
-							// the last packet will contain the actual amount retrieved, not SQL_NO_TOTAL
-							// so we calculate the actual length of the string with that.
-							if( dummy_field_len != SQL_NO_TOTAL )
-								field_len_temp += dummy_field_len;
-							else
-								field_len_temp += initial_field_len;
-
-							if( r == SQL_SUCCESS_WITH_INFO ) {
-								core::SQLGetDiagField( stmt, 1, SQL_DIAG_SQLSTATE, state, SQL_SQLSTATE_BUFSIZE, &len
-													   TSRMLS_CC );
-							}
-
-						} while( r == SQL_SUCCESS_WITH_INFO && is_truncated_warning( state ));
-					}
-					else {
-
-						// We got the field_len_temp from SQLGetData call.
-						field_value_temp = static_cast<char*>( sqlsrv_realloc( field_value_temp, field_len_temp + extra + 1 ));
-
-						// We have already recieved INITIAL_FIELD_STRING_LEN size data.
-						field_len_temp -= INITIAL_FIELD_STRING_LEN;
```

Comment:
Looks like the block (lines 2261 to 2269) is the same as the code in the else block (lines 2274 to 2282)?
If so, is it possible to refactor the code?